docs/user-guide.md

```diff
@@ -0,0 +1,659 @@
+--cors-exposes-headers [--cors-exposes-headers option]  set the expose cors headers access control (Access-Control-Expose-Headers)
+```
+
+## Upstream URL
+
+You can control the upstream endpoint via the `--upstream-url` option.
+Both HTTP and HTTPS are supported with TLS verification and keep-alive
+support configured via the `--skip-upstream-tls-verify` /
+`--upstream-keepalives` option. Note, the proxy can also upstream via a
+UNIX socket, `--upstream-url unix://path/to/the/file.sock`.
+
+## Endpoints
+
+  - **/oauth/authorize** is authentication endpoint which will generate
+    the OpenID redirect to the provider
+
+  - **/oauth/callback** is provider OpenID callback endpoint
+
+  - **/oauth/expired** is a helper endpoint to check if a access token
+    has expired, 200 for ok and, 401 for no token and 401 for expired
+
+  - **/oauth/health** is the health checking endpoint for the proxy, you
+    can also grab version from headers
+
+  - **/oauth/login** provides a relay endpoint to login via
+    `grant_type=password`, for example, `POST /oauth/login` form values
+    are `username=USERNAME&password=PASSWORD` (must be enabled)
+
+  - **/oauth/logout** provides a convenient endpoint to log the user
+    out, it will always attempt to perform a back channel log out of
+    offline tokens
+
+  - **/oauth/token** is a helper endpoint which will display the current
+    access token for you
+
+  - **/oauth/metrics** is a Prometheus metrics handler
+
+## Metrics
+
+Assuming `--enable-metrics` has been set, a Prometheus endpoint can be
+found on **/oauth/metrics**; at present the only metric being exposed is
+a counter per HTTP code.
+
+## Limitations
+
+Keep in mind [browser cookie
+limits](http://browsercookielimits.squawky.net/) if you use access or
+refresh tokens in the browser cookie. Louketo Proxy divides
+the cookie automatically if your cookie is longer than 4093 bytes. The real
+size of the cookie depends on the content of the issued access token.
+Also, encryption might add additional bytes to the cookie size. If you
+have large cookies (\>200 KB), you might reach browser cookie limits.
+
+All cookies are part of the header request, so you might find a problem
+with the max headers size limits in your infrastructure (some load
+balancers have very low this value, such as 8 KB). Be sure that all
+network devices have sufficient header size limits. Otherwise, your
+users won’t be able to obtain an access token.
+
+## Known Issues
+
+There is a known issue with the Keycloak server 4.6.0.Final in which
+Louketo Proxy is unable to find the *client\_id* in the *aud* claim. This
+is due to the fact the *client\_id* is not in the audience anymore. The
+workaround is to add the "Audience" protocol mapper to the client with
+the audience pointed to the *client\_id*. For more information, see
+[KEYCLOAK-8954](https://issues.redhat.com/browse/KEYCLOAK-8954).
```

Comment:
Note about historic names should be mentioned in the README: gambol99/keycloak-proxy, keycloak/keycloak-gatekeeper. Users may search them and they may find historic problems/comments which may help them with their current issues.

Reply:
Hi, @jangaraj what exactly you would like to suggest? 

If you search for `gambol99/keycloak-proxy` or `keycloak/keycloak-gatekeeper` the search returns louketo-proxy. Could you please add the changes you mentioned here or into the README. I will be more than happy to incorporate it.

Reply:
Sure, GitHub repo is redirected correctly. But there is existing knowledge, which points to that old names, see Google: `keycloak/keycloak-gatekeeper site:stackoverflow.com`. There is even dedicated label on stackoverlow. It will be nice to have reference to that old names in the readme, so users will know that any details about `keycloak-gatekeeper/keycloak-proxy` may be relevant also to `louketo-proxy`.

Reply:
@jangaraj I still need to talk about this with @stianst, but one idea is to add this in the Keycloak documentation for Gatekeeper, mentioning that we moved the repository to the Louketo organization. So in the end, adding this, might not be necessary, but I'm not against it.